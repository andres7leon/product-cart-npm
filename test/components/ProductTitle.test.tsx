import React from 'react'
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';


const product1 = {
    id: '1',
    title: 'Coffee Mug - Card',
    img: './coffee-mug.png'
}

// const product2 = {
//     id: '2',
//     title: 'Cafeee normal',
//     img: './coffee-mug.png'
// }

describe('ProductTitle', () => { 
    
    test('debe de mostrar el componente correctaent con el titulo personalizado', () => {
        
        const wrapper = renderer.create(
            <ProductTitle title='Custom Product' />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();

    })

    test('debe de mostrar el componente con el nombre del producto', () => { 

        const wrapper = renderer.create(
            <ProductCard product={ product1 } >
                {
                    () => (
                        <ProductTitle  title='es un texto'/>
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
        
    })
})