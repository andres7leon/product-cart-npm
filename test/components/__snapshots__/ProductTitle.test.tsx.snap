// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    es un texto
  </span>
</div>
`;

exports[`ProductTitle debe de mostrar el componente correctaent con el titulo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;
